src/FakeItEasy/Core/WrappedObjectRule.cs

```diff
@@ -50,8 +58,32 @@ public void Apply(IInterceptedFakeObjectCall fakeObjectCall)
             Guard.AgainstNull(fakeObjectCall, nameof(fakeObjectCall));
 
             var parameters = fakeObjectCall.Arguments.GetUnderlyingArgumentsArray();
-            var valueFromWrappedInstance = fakeObjectCall.Method.Invoke(this.wrappedObject, parameters);
-            fakeObjectCall.SetReturnValue(valueFromWrappedInstance);
+            try
+            {
+                var valueFromWrappedInstance = fakeObjectCall.Method.Invoke(this.wrappedObject, parameters);
+                fakeObjectCall.SetReturnValue(valueFromWrappedInstance);
+            }
+            catch (TargetInvocationException ex)
+            {
+                RethrowPreservingStackTrace(ex.InnerException);
+            }
+        }
+
+        private static void RethrowPreservingStackTrace(Exception e)
+        {
+#if FEATURE_BINARY_SERIALIZATION
+            var ctx = new StreamingContext(StreamingContextStates.CrossAppDomain);
+            var mgr = new ObjectManager(null, ctx);
+            var si = new SerializationInfo(e.GetType(), new FormatterConverter());
+
+            e.GetObjectData(si, ctx);
+            mgr.RegisterObject(e, 1, si); // prepare for SetObjectData
+            mgr.DoFixups(); // ObjectManager calls SetObjectData
+
+            throw e;
+#else
+            ExceptionDispatchInfo.Capture(e).Throw();
+#endif

```

Comment:
Ah, I thought we must already have something like that, but didn't find it. Thanks!
However, it works on the full .NET Framework, but probably not on .NET Core... We must have missed it during the port to .NET Core because of the empty catch. I'll extract it to a separate class and make it work for .NET Core.